### ISSUE TITLE
Bridge should report the number of users "using" the bridge via metrics.

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/matrix-appservice-slack#378]
For Matrix users, this should be defined by the following statements:
- They have sent a message/joined a room/caused some action to occur on the bridge.
- They have been active for at least N days in a given month. For example, talking on the 1st and then again on the 28th would consider them active as the timestamp between their first interation and their last interaction is significant. However, someone who only talked once is not to be considered active.
- OR they have an account that is puppeted by the bridge.

For slack users, this should be:
- They have sent a message/joined a channel/caused some action to occur on the bridge.
- They have been active for at least N days in a given month. For example, talking on the 1st and then again on the 28th would consider them active as the timestamp between their first interation and their last interaction is significant. However, someone who only talked once is not to be considered active.
- We should ignore slack bot users for now.

And just for fun, we should include the number of bridged rooms, again by activity:
- How many rooms have seen action, within a given period.

The metrics should include:
- How many active matrix users NOT being puppeted
- How many active matrix users being puppeted.
- How many rooms are active, by type of room.

**Comment by jaller94** [comment]
**Work in progress**

New config variable:
```
# Amount of days that a user or a room needs to have bridged events to count as active
metrics_active_days_threshold: 2
```

**All gauges also have a label "team_id"**
```
# HELP bridge_active_users Amount of active users
# TYPE bridge_active_users gauge
bridge_active_users{remote=false} 34
bridge_active_users{remote=true} 27

# HELP bridge_user_puppets Amount of puppeted users on the remote side of the bridge
# TYPE bridge_user_puppets gauge
bridge_user_puppets 12

# HELP bridge_rooms Amount of active, bridged rooms per team (types are "channel" and "user")
# TYPE bridge_rooms gauge
bridge_active rooms{type="channel"} 20
bridge_active_rooms{remote="user"} 51
```

**Comment by jaller94** [comment]
Solved in #380.
